deps/verifier/src/snp/mod.rs

```diff
@@ -84,8 +92,14 @@ impl Verifier for Snp {
             cert_chain,
         } = serde_json::from_slice(evidence).context("Deserialize Quote failed.")?;
 
-        let Some(cert_chain) = cert_chain else {
-            bail!("Cert chain is unset");
+        let cert_chain = if let Some(chain) = cert_chain {
+            if chain.is_empty() {
+                fetch_vcek_from_kds(report)?
+            } else {
+                chain
+            }
+        } else {
+            fetch_vcek_from_kds(report)?
```

Comment:
I think you can simplify this by doing something like 

```
let cert_chain = match cert_chain {
    Some(chain) if !chain.is_empty() => chain,
    _ => fetch_vcek_from_kds(report)?
}
```
Or something like that

Reply:
Fixed!